### DISCUSSION TITLE
FileNotFoundError: [Errno 2] No such file or directory:

### DISCUSSION THREAD

**Discussion by safdarianebi** [su2code/SU2#2205] [General]
Hi,
I'm using SU2 version 8.0.0 and I want to run a case related to axial compressor passage, but unfortunately I'm receiving this error:

##############################################################################################
Info    : Done meshing 3D (Wall 1.78e-05s, CPU 0s)
Info    : 6934 nodes 10367 elements
Setting up SU2 Simulation for /home/ebi/Master-Thesis/simulation_out/config1.cfg
Running SU2 Simulation for /home/ebi/Master-Thesis/simulation_out/config1.cfg

Error in "void CConfig::SetConfig_Parsing(std::istream&)": 
-------------------------------------------------------------------------
Line 78 TURBO_PERF_KIND: invalid option name. Check current SU2 options in config_template.cfg.
Did you mean TURBOMACHINERY_KIND?

------------------------------ Error Exit -------------------------------

2024-01-27 00:36:54.308 (2338.855s) [         A720740]       vtkXMLReader.cxx:306    ERR| vtkXMLUnstructuredGridReader (0x7fffcb20e9d0): Error opening file /home/ebi/Master-Thesis/simulation_out/flow1_0.vtu
2024-01-27 00:36:54.308 (2338.856s) [         A720740]       vtkExecutive.cxx:740    ERR| vtkCompositeDataPipeline (0x7fffcbbd3850): Algorithm vtkXMLUnstructuredGridReader (0x7fffcb20e9d0) returned failure for request: vtkInformation (0x7fffcbf4b550)
  Debug: Off
  Modified Time: 1357
  Reference Count: 1
  Registered Events: (none)
  Request: REQUEST_INFORMATION
  ALGORITHM_AFTER_FORWARD: 1
  FORWARD_DIRECTION: 0

2024-01-27 00:36:54.308 (2338.856s) [         A720740]       vtkXMLReader.cxx:306    ERR| vtkXMLUnstructuredGridReader (0x7fffcb20e9d0): Error opening file /home/ebi/Master-Thesis/simulation_out/flow1_1.vtu
2024-01-27 00:36:54.308 (2338.856s) [         A720740]       vtkExecutive.cxx:740    ERR| vtkCompositeDataPipeline (0x7fffcbbd3850): Algorithm vtkXMLUnstructuredGridReader (0x7fffcb20e9d0) returned failure for request: vtkInformation (0x7fffcbebfb60)
  Debug: Off
  Modified Time: 1456
  Reference Count: 1
  Registered Events: (none)
  Request: REQUEST_INFORMATION
  ALGORITHM_AFTER_FORWARD: 1
  FORWARD_DIRECTION: 0

---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[9], line 5
      2 from paraflow.simulation.su2 import Su2SimulationConfig 
      4 flasher = get_flasher("air", "gas")
----> 5 sim_results = run_simulation(
      6     axial_turbo_passage,
      7     params=SimulationParams(
      8         inlet_total_state=flasher.flash(P=87909, T=276, mach_number=0.4098), 
      9         target_outlet_static_state=flasher.flash(P=108932, T=296),
     10         translation=[
     11             np.array([0.0, 266, 0.0]),
     12             None
     13         ]
     14     ),
     15     working_directory="./simulation_out", 
     16     id="1",
     17     auto_delete=True,
     18     verbose=True,
     19     # num_procs=9,
     20     cfg=Su2SimulationConfig(
     21         custom_repo_url= "https://github.com/su2code/SU2/tree/master",
     22         version= "8.0.0"
     23     )
     24 )
     25 sim_results.to_file("./simulation_out/cached.pkl")
     27 # sim_results = SimulationResult.from_file("./simulation_out/cached.pkl")

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/paraflow/simulation/simulation.py:21, in run_simulation(passage, params, working_directory, id, auto_delete, verbose, num_procs, cfg)
     19 if isinstance(cfg, Su2SimulationConfig):
     20     from paraflow.simulation.su2 import run_su2_simulation
---> 21     sim_results = run_su2_simulation(passage, params, working_directory, id, auto_delete, verbose, num_procs, cfg)
     23 return sim_results

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/paraflow/simulation/su2.py:161, in run_su2_simulation(passage, params, working_directory, id, auto_delete, verbose, num_procs, cfg)
    158         vtu_filename = vtu_filename.replace(".", f"_{i}.")
    160     grids += [read_vtu_data(vtu_filename)]
--> 161 eval_values = pd.read_csv(config["CONV_FILENAME"])
    164 if auto_delete:
    165     delete_output_files(config, config_path, num_zones)

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/pandas/io/parsers/readers.py:1024, in read_csv(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)
   1011 kwds_defaults = _refine_defaults_read(
   1012     dialect,
   1013     delimiter,
   (...)
   1020     dtype_backend=dtype_backend,
   1021 )
   1022 kwds.update(kwds_defaults)
-> 1024 return _read(filepath_or_buffer, kwds)

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/pandas/io/parsers/readers.py:618, in _read(filepath_or_buffer, kwds)
    615 _validate_names(kwds.get("names", None))
    617 # Create the parser.
--> 618 parser = TextFileReader(filepath_or_buffer, **kwds)
    620 if chunksize or iterator:
    621     return parser

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/pandas/io/parsers/readers.py:1618, in TextFileReader.__init__(self, f, engine, **kwds)
   1615     self.options["has_index_names"] = kwds["has_index_names"]
   1617 self.handles: IOHandles | None = None
-> 1618 self._engine = self._make_engine(f, self.engine)

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/pandas/io/parsers/readers.py:1878, in TextFileReader._make_engine(self, f, engine)
   1876     if "b" not in mode:
   1877         mode += "b"
-> 1878 self.handles = get_handle(
   1879     f,
   1880     mode,
   1881     encoding=self.options.get("encoding", None),
   1882     compression=self.options.get("compression", None),
   1883     memory_map=self.options.get("memory_map", False),
   1884     is_text=is_text,
   1885     errors=self.options.get("encoding_errors", "strict"),
   1886     storage_options=self.options.get("storage_options", None),
   1887 )
   1888 assert self.handles is not None
   1889 f = self.handles.handle

File ~/miniforge3/envs/cadquery/lib/python3.11/site-packages/pandas/io/common.py:873, in get_handle(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)
    868 elif isinstance(handle, str):
    869     # Check whether the filename is to be opened in binary mode.
    870     # Binary mode does not support 'encoding' and 'newline'.
    871     if ioargs.encoding and "b" not in ioargs.mode:
    872         # Encoding
--> 873         handle = open(
    874             handle,
    875             ioargs.mode,
    876             encoding=ioargs.encoding,
    877             errors=errors,
    878             newline="",
    879         )
    880     else:
    881         # Binary mode
    882         handle = open(handle, ioargs.mode)

FileNotFoundError: [Errno 2] No such file or directory: '/home/ebi/Master-Thesis/simulation_out/config1.csv'
​
##############################################################################################

Your guidance on this matter is highly appreciated.

**Comment by safdarianebi** [comment]
The code that encounters this error is as below:

##############################################################################################
import numpy as np
from parafoil import TurboRowPassage, TurboStagePassage, TurboMeshParameters
from parafoil import CamberThicknessAirfoil

rotor_airfoil = CamberThicknessAirfoil(
    inlet_angle=np.radians(-60),
    outlet_angle=np.radians(-51),
    upper_thick_prop=[0.015, 0.05, 0.05  , 0.015],
    lower_thick_prop=[0.015, 0.05, 0.05, 0.015],
    leading_prop=0.5,
    trailing_prop=0.5,
    chord_length=0.018,
)

stator_airfoil = CamberThicknessAirfoil(
    inlet_angle=np.radians(26),
    outlet_angle=np.radians(12),
    upper_thick_prop=[0.015, 0.05, 0.05  , 0.015],
    lower_thick_prop=[0.015, 0.05, 0.05, 0.015],
    leading_prop=0.5,
    trailing_prop=0.5,
    chord_length=0.018,
)

axial_turbo_passage = TurboStagePassage(
    inflow_passage=TurboRowPassage(
        airfoil=rotor_airfoil,
        spacing_to_chord=0.5,
        leading_edge_gap_to_chord=1.0,
        trailing_edge_gap_to_chord=0.25,
        type="camber",
        mesh_params=TurboMeshParameters(
            airfoil_label="wall1",
            bottom_label="periodic1",
            top_label="periodic2",
            inlet_label="inflow",
            outlet_label="outmix",
        )
    ),
    outflow_passage=TurboRowPassage(
        airfoil=stator_airfoil,
        spacing_to_chord=0.5,
        leading_edge_gap_to_chord=0.25,
        trailing_edge_gap_to_chord=1.0,
        type="camber",
        mesh_params=TurboMeshParameters(
            airfoil_label="wall2",
            bottom_label="periodic3",
            top_label="periodic4",
            inlet_label="inmix",
            outlet_label="outflow",
        )
    )
)

##############################################################################################

from paraflow import run_simulation, get_flasher, SimulationParams, get_frames, display_frame
from paraflow.simulation.su2 import Su2SimulationConfig 

flasher = get_flasher("air", "gas")
sim_results = run_simulation(
    axial_turbo_passage,
    params=SimulationParams(
        inlet_total_state=flasher.flash(P=87909, T=276, mach_number=0.4098), 
        target_outlet_static_state=flasher.flash(P=108932, T=296),
        translation=[
            np.array([0.0, 266, 0.0]),
            None
        ]
    ),
    working_directory="./simulation_out", 
    id="1",
    auto_delete=True,
    verbose=True,
    # num_procs=9,
    cfg=Su2SimulationConfig(
        custom_repo_url= "https://github.com/su2code/SU2/tree/master",
        version= "8.0.0"
    )
)
sim_results.to_file("./simulation_out/cached.pkl")

# sim_results = SimulationResult.from_file("./simulation_out/cached.pkl")

frames = get_frames(
    sim_results, 
    property_names=["Pressure"], 
    num_pnts=500,
)
display_frame(frames, "Pressure")
sim_results.eval_values.loc[sim_results.eval_values.index[-1], '  "TotTotEff[1]"  ']

**Comment by bigfooted** [comment]
> Line 78 TURBO_PERF_KIND: invalid option name. Check current SU2 options in config_template.cfg.

I am not so familiar with the turbomachinery implementation so I hope somebody else will chime in, but this option was deprecated. Maybe have a look at the testcases/turbomachinery folder that we have, there might be a testcase that matches what you would like to do, maybe transonic_stator_2D?
